rskj-core/src/main/java/co/rsk/remasc/Remasc.java

```diff
@@ -116,6 +123,39 @@ void processMinersFees() {
         feesPayer.payMiningFees(processingBlockHeader.getHash(), payToRskLabs, remascConstants.getRskLabsAddress(), logs);
         fullBlockReward = fullBlockReward.subtract(payToRskLabs);
 
+        // TODO to improve
+        // this type choreography is only needed because the RepositoryTrack support the
+        // get snapshot to method
+        Repository processingRepository = ((RepositoryTrack)repository).getOriginRepository().getSnapshotTo(processingBlockHeader.getStateRoot());
+        // TODO to improve
+        // and we need a RepositoryTrack to feed RemascFederationProvider
+        // because it supports the update of bytes (notably, RepositoryImpl don't)
+        // the update of bytes is needed, because BridgeSupport creation could alter
+        // the storage when getChainHead is null (specially in production)
+        processingRepository = processingRepository.startTracking();
+        RemascFederationProvider federationProvider = new RemascFederationProvider(processingRepository);
+
+        BigInteger payToFederation = fullBlockReward.divide(BigInteger.valueOf(remascConstants.getFederationDivisor()));
+
+        byte[] processingBlockHash = processingBlockHeader.getHash();
+        int nfederators = federationProvider.getFederationSize();
+        BigInteger payToFederator = payToFederation.divide(BigInteger.valueOf(nfederators));
+        BigInteger restToLastFederator = payToFederation.subtract(payToFederator.multiply(BigInteger.valueOf(nfederators)));
+        BigInteger paidToFederation = BigInteger.ZERO;
+
+        for (int k = 0; k < nfederators; k++) {
+            byte[] federatorAddress = federationProvider.getFederatorAddress(k);
+
+            if (k == nfederators - 1 && restToLastFederator.compareTo(BigInteger.ZERO) > 0)
+                feesPayer.payMiningFees(processingBlockHash, payToFederator.add(restToLastFederator), federatorAddress, logs);
```

Comment:
The uploaded test do not reach here ever. 
TODO: Get a test that ensures that the last member of the federation has a little more fees.
IMPROVEMENT: Random the last member of the federation? To make is fair